### PR TITLE
Use CMake build system

### LINKED ISSUE #9: Fix tests/bandagetests.cpp
1) The signature of the getStartingNodes function has changed. The test is not compiled.
2) linker problems
...

### PR DESCRIPTION
**Pull request by wafemand**
Fix issue https://github.com/asl/Bandage/issues/1

Partially fix https://github.com/asl/Bandage/issues/9 (build errors only)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+set(CMAKE_AUTOMOC ON)
+set(CMAKE_AUTORCC ON)
+set(CMAKE_AUTOUIC ON)
+
+find_package(Qt6 REQUIRED COMPONENTS Widgets Svg Test)
+
+set(LIB_SOURCES
+        program/settings.cpp
```
Comment (asl) [inline CMakeLists.txt]: Please sort by dir, so we will be able to split it further on by libraries

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        ui/blasthitfiltersdialog.ui
+        ui/changenodenamedialog.ui
+        ui/graphinfodialog.ui
+        ui/changenodedepthdialog.ui)
+
+set(RESOURCES images/images.qrc)
+
+add_compile_definitions(QT_DISABLE_DEPRECATED_BEFORE=0x050F00)
+add_compile_definitions(APP_VERSION="0.9.0-dev")
+include_directories(.)
```
Comment (asl) [inline CMakeLists.txt]: https://cmake.org/cmake/help/latest/variable/CMAKE_INCLUDE_CURRENT_DIR.html ?

File: CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        ui/querypathsdialog.ui
+        ui/blasthitfiltersdialog.ui
+        ui/changenodenamedialog.ui
+        ui/graphinfodialog.ui
+        ui/changenodedepthdialog.ui)
+
+set(RESOURCES images/images.qrc)
+
+add_compile_definitions(QT_DISABLE_DEPRECATED_BEFORE=0x050F00)
+add_compile_definitions(APP_VERSION="0.9.0-dev")
```
Comment (asl) [inline CMakeLists.txt]: Can we have some single place for setting the version? E.g. one single variable?

File: blast/blasthit.cpp [hunk 4]
```diff
@@ -18,10 +18,10 @@
 
 #include "blasthit.h"
 
-#include "../graph/debruijnnode.h"
```
Comment (asl) [inline blast/blasthit.cpp:21]: Can we somehow sort includes while here?

E.g.

"local ones"
"external, but local to project"
"QT"
"C++ standard"
"C standard"

Comment (wafemand) [reply blast/blasthit.cpp:21]: Suggest doing that in the MR fixing code style. I'll set up clang-format and format check in the pipeline.

Now I don't know how to sort includes automatically with clang-format without reformatting the whole project source )

**Pull request merged**